cwltool/workflow_job.py

```diff
@@ -386,6 +386,8 @@ def object_from_state(
     incomplete: bool = False,
 ) -> Optional[CWLObjectType]:
     inputobj = {}  # type: CWLObjectType
+    for s in state.keys():
+        inputobj[s] = state[s].value
```

Comment:
I think we can avoid accessing keys & values separately. Something like:

```python
for k, v in state.items():
  inputobj[k] = v
```

Or start `inputobj` by deep-cloning `state`.

Reply:
Current commited version is 
https://github.com/common-workflow-language/cwltool/pull/1566/files#diff-ed87d61e2237460638e7f3ddb21bafc9969d8c142c804d79f94801142342f3d6R389-R391

My local version is following .
But I can not pass `py39-mypy`

```python
    for s in state.keys():
        if hasattr(state[s], "value"):
            inputobj[s] = state[s].value
```

I replace it to your code.

```
    for k,v  in state.items():
        if type(v) is WorkflowStateItem:
            if hasattr(v, "value"):
                inputobj[k] = v.value
```